packages/core/src/diff/enum.ts

```diff
@@ -11,63 +15,106 @@ import {
     },
     onRemoved(value) {
-      addChange(enumValueRemoved(oldEnum, value));
+      addChange(enumValueRemoved(oldEnum!, value));
     },
     onMutual(value) {
-      const oldValue = value.oldVersion;
-      const newValue = value.newVersion;
-
-      if (isNotEqual(oldValue.description, newValue.description)) {
-        addChange(enumValueDescriptionChanged(newEnum, oldValue, newValue));
-      }
-
-      if (isNotEqual(oldValue.deprecationReason, newValue.deprecationReason)) {
-        if (isVoid(oldValue.deprecationReason)) {
-          addChange(enumValueDeprecationReasonAdded(newEnum, oldValue, newValue));
-        } else if (isVoid(newValue.deprecationReason)) {
-          addChange(enumValueDeprecationReasonRemoved(newEnum, oldValue, newValue));
-        } else {
-          addChange(enumValueDeprecationReasonChanged(newEnum, oldValue, newValue));
-        }
-      }
-
-      compareLists(oldValue.astNode?.directives || [], newValue.astNode?.directives || [], {
-        onAdded(directive) {
-          addChange(
-            directiveUsageAdded(Kind.ENUM_VALUE_DEFINITION, directive, {
-              type: newEnum,
-              value: newValue,
-            }),
-          );
-        },
-        onRemoved(directive) {
-          addChange(
-            directiveUsageRemoved(Kind.ENUM_VALUE_DEFINITION, directive, {
-              type: oldEnum,
-              value: oldValue,
-            }),
-          );
-        },
-      });
+      changesInEnumValue(value, newEnum, addChange);
     },
   });
 
-  compareLists(oldEnum.astNode?.directives || [], newEnum.astNode?.directives || [], {
+  compareLists(oldEnum?.astNode?.directives || [], newEnum.astNode?.directives || [], {
     onAdded(directive) {
-      addChange(directiveUsageAdded(Kind.ENUM_TYPE_DEFINITION, directive, newEnum));
+      addChange(
+        directiveUsageAdded(Kind.ENUM_TYPE_DEFINITION, directive, newEnum, oldEnum === null),
+      );
+      directiveUsageChanged(null, directive, addChange, newEnum);
+    },
+    onMutual(directive) {
+      directiveUsageChanged(directive.oldVersion, directive.newVersion, addChange, newEnum);
     },
     onRemoved(directive) {
       addChange(directiveUsageRemoved(Kind.ENUM_TYPE_DEFINITION, directive, newEnum));
     },
   });
 }
+
+function changesInEnumValue(
+  value: {
+    newVersion: GraphQLEnumValue;
+    oldVersion: GraphQLEnumValue | null;
+  },
+  newEnum: GraphQLEnumType,
+  addChange: AddChange,
+) {
+  const oldValue = value.oldVersion;
+  const newValue = value.newVersion;
+
+  if (isNotEqual(oldValue?.description, newValue.description)) {
+    addChange(enumValueDescriptionChanged(newEnum, oldValue, newValue));
+  }
+
+  if (isNotEqual(oldValue?.deprecationReason, newValue.deprecationReason)) {
+    if (
+      isVoid(oldValue?.deprecationReason) ||
+      oldValue?.deprecationReason === DEPRECATION_REASON_DEFAULT
+    ) {
+      addChange(enumValueDeprecationReasonAdded(newEnum, oldValue, newValue));
```

Comment:
I think it's strange that not all types check for deprecationReason... 
Thoughts on removing this in favor of only using the directive change for `@deprecated`?

Reply:
You mean like in general no longer have deprecation changes? If so there might be some value in treating deprecations as something special/separate. 🤔  But it could always be computed from the directive changes 🤔

Reply:
yeah.

For now I've left these changes in place but I'm also emitting a generic directive change. And within patch, I'm only using the generic directive change to compute the patch.